contracts/LiquidDemocracy.sol

```diff
@@ -56,6 +57,59 @@ contract LiquidDemocracy {
 
         votersData[voterAddress].weight = 1;
         votersData[voterAddress].registered = true;
+        delegations[voterAddress] = voterAddress;
+    }
+
+    // Delegations
+
+    /**
+     * Delegate the voting power of a voter to another voter
+     * Only the person calling this function has the power of delegating his/her vote
+     * @param representative voter to delegate the voting power to
+     */
+    function delegate(address representative) public {
+
+        address representee = msg.sender;
+        require(delegations[representee] == representee); // At the moment you can apply a delegation once
+        require(votersData[representee].registered);
+        require(votersData[representative].registered);
+
+        delegations[representee] = representative; // Delegate my own vote
+
+        address nextRepresentative = representative;
+        address nextRepresentee = representee;
+
+        // While there is voter that hasn't performed a delegation to another voter
+        // transitively add votes
+        while (delegations[nextRepresentative] != nextRepresentative) {
+
+            votersData[nextRepresentative].weight += votersData[nextRepresentee].weight;
+            votersData[nextRepresentee].weight -= votersData[nextRepresentee].weight;
+
+            nextRepresentee = nextRepresentative;
+            nextRepresentative = delegations[nextRepresentative];
+
+        }
+
+        votersData[nextRepresentative].weight += votersData[nextRepresentee].weight;
+        votersData[nextRepresentee].weight -= votersData[nextRepresentee].weight;
+
+    }
+
+    /**
+    * Gets the weight of the voter that calls this method
+    */
+    function getMyWeight() public view returns (uint) {
+        require(votersData[msg.sender].registered);
```

Comment:
This `require()` for getters makes sense, wondering if we should include for `getVotersData()` as well

Reply:
I'll add it

Reply:
actually, thinking it better, a property of getVotersData is to return if a "voter" is registered or not. But I see no usage or use case of this function yet, should we just remove it instead?

Reply:
Ah yea you're right, probably best to leave as is for now